drivers/ipm/ipm_mhu.h

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2019 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_DRIVERS_IPM_IPM_MHU_H_
+#define ZEPHYR_DRIVERS_IPM_IPM_MHU_H_
+
+#include <kernel.h>
+#include <ipm.h>
+#include <device.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define IPM_MHU_MAX_DATA_SIZE		1
+#define IPM_MHU_MAX_ID_VAL		0
+#define SSE_200_CPU_ID_UNIT_OFFSET	((0x1F000UL))
+
+/* SSE 200 MHU register map structure */
+struct ipm_mhu_reg_map_t {
```

Comment:
Are these definitions somewhere in SoC's vendor SDK? If so, the vendor headers could be put into ext/hal/... and included and reused here (assuming there are no license issues).

Reply:
The ARM IP has followed this model of not having a SDK or headers, so its fine that we define the register map here.

Reply:
Ok. Btw I think checkpatch.pl ignores the ext folder?